lib/serialization/ObjectMiddleware.js

```diff
@@ -46,11 +46,13 @@ Technically any value can be used.
 /**
  * @typedef {Object} ObjectSerializerContext
  * @property {function(any): void} write
+ * @property {function(any): void} setCircularReference
  */
 
 /**
  * @typedef {Object} ObjectDeserializerContext
  * @property {function(): any} read
+ * @property {function(any): void} setCircularReference
```

Comment:
The funny thing is that we use this in some methods, but it was not in the types, and for some reason typescript doesn't catch it

Reply:
cc @danielrosenwasser

Reply:
Just to be clear, you're saying that before this change, uses of this method were not caught, and that's the bug? Just trying to understand what is is to be testing.

Reply:
@jakebailey Yeah, this is not a bug, but rather an improvement, just reported that this problem was not caught before

Reply:
Ah, gotcha.